src/FEValues/common_values.jl

```diff
@@ -211,6 +211,41 @@ function function_gradient_init(cv::AbstractValues, ::AbstractVector{T}) where {
     return zero(T) ⊗ zero(shape_gradient_type(cv))
 end
 
+"""
+    function_hessian(fe_v::AbstractValues{dim}, q_point::Int, u::AbstractVector{<:AbstractFloat}, [dof_range])
+
+    Compute the hessian of the function in a quadrature point. `u` is a vector with values
+    for the degrees of freedom. 
+"""
+function function_hessian(fe_v::AbstractValues, q_point::Int, u::AbstractVector, dof_range = eachindex(u))
+    n_base_funcs = getnbasefunctions(fe_v)
+    length(dof_range) == n_base_funcs || throw_incompatible_dof_length(length(dof_range), n_base_funcs)
+    @boundscheck checkbounds(u, dof_range)
+    @boundscheck checkquadpoint(fe_v, q_point)
+    hess = function_hessian_init(fe_v, u)
+    @inbounds for (i, j) in pairs(dof_range)
+        hess += shape_hessian(fe_v, q_point, i) * u[j]
+    end
+    return hess
+end
+
+"""
+    shape_hessian_type(fe_v::AbstractValues)
+
+Return the type of `shape_hessian(fe_v, q_point, base_function)`
+"""
+function shape_hessian_type(fe_v::AbstractValues)
+    # Default fallback
+    return typeof(shape_hessian(fe_v, 1, 1))
+end
+
+function function_hessian_init(cv::AbstractValues, ::AbstractVector{T}) where {T}
+    return zero(shape_hessian_type(cv)) * zero(T)
+end
+function function_hessian_init(cv::AbstractValues, ::AbstractVector{T}) where {T <: AbstractVector}
+    return zero(T) ⊗ zero(shape_hessian_type(cv))
+end
```

Comment:
```suggestion
```
`T <: AbstractVector` is not compatible with the `function_hessian` above and this isn't covered by tests, so I guess unused?